ang/test/civicase/Search.spec.js

```diff
@@ -0,0 +1,398 @@
+        beforeEach(function () {
+          event.type = 'click';
+        });
+
+        describe('when isDeleted is true', function () {
+          beforeEach(function () {
+            element.isolateScope().filters.is_deleted = true;
+            element.isolateScope().toggleIsDeleted(event);
+          });
+
+          it('isDeleted to be false', function () {
+            expect(element.isolateScope().filters.is_deleted).toBe(false);
+          });
+
+          it('event.preventDefault to be called', function () {
+            expect(event.preventDefault).toHaveBeenCalled();
+          });
+        });
+
+        describe('when isDeleted is false', function () {
+          beforeEach(function () {
+            element.isolateScope().filters.is_deleted = false;
+            element.isolateScope().toggleIsDeleted(event);
+          });
+
+          it('isDeleted to be true', function () {
+            expect(element.isolateScope().filters.is_deleted).toBe(true);
+          });
+
+          it('event.preventDefault to be called', function () {
+            expect(event.preventDefault).toHaveBeenCalled();
+          });
+        });
+      });
+
+      describe('keydown event triggered with enter key', function () {
+        beforeEach(function () {
+          event.type = 'keydown';
+          event.keyCode = 13;
+        });
+
+        describe('when isDeleted is true', function () {
+          beforeEach(function () {
+            element.isolateScope().filters.is_deleted = true;
+            element.isolateScope().toggleIsDeleted(event);
+          });
+
+          it('isDeleted to be false', function () {
+            expect(element.isolateScope().filters.is_deleted).toBe(false);
+          });
+
+          it('event.preventDefault to be called', function () {
+            expect(event.preventDefault).toHaveBeenCalled();
+          });
+        });
+
+        describe('when isDeleted is false', function () {
+          beforeEach(function () {
+            element.isolateScope().filters.is_deleted = false;
+            element.isolateScope().toggleIsDeleted(event);
+          });
+
+          it('isDeleted to be true', function () {
+            expect(element.isolateScope().filters.is_deleted).toBe(true);
+          });
+
+          it('event.preventDefault to be called', function () {
+            expect(event.preventDefault).toHaveBeenCalled();
+          });
+        });
+      });
+
+      describe('keydown event triggered with space key', function () {
+        beforeEach(function () {
+          event.type = 'keydown';
+          event.keyCode = 32;
+        });
+
+        describe('when isDeleted is true', function () {
+          beforeEach(function () {
+            element.isolateScope().filters.is_deleted = true;
+            element.isolateScope().toggleIsDeleted(event);
+          });
+
+          it('isDeleted to be false', function () {
+            expect(element.isolateScope().filters.is_deleted).toBe(false);
```

Comment:
this is not how directives testing should be done, usually. When testing directives you do compile the element, but you trigger events using `element.find('child-element').trigger('keydown', { which: 'Enter' })` and test the HTML outcome, not the scope. Ex: `expect(element.find('.is-deleted-element:visible').length).toBe(1)` or similar.

I'd only use this kind of testing when simulating the event is too complicated (scrolling, dragging things, etc.).